### ISSUE TITLE
EmbeddedDocument @id changes after every merge()

### ISSUE THREAD

**Issue by websirnik** [doctrine/mongodb-odm#1297]
I'm seeing a strange behaviour atm. I have a document with embedded document. I want to have an id for every EmbeddedDocument. Ids get generated after the document gets flushed initially. But every time I merge() the document to update it, every EmbeddedDocument gets assigned new ids. This is not what I expect. Is it a bug? If now, how do I keep the same Ids after merge()?

```php
/**
 * @MongoDB\Document
 */
class Post
{
    /* ... */

    /**
     * @MongoDB\EmbedMany(targetDocument="Item")
     */
    protected $items = array();
}

```

```php
/**
 * @MongoDB\EmbeddedDocument
 */
class Item
{
    /**
     * @MongoDB\Id(strategy="AUTO")
     */
    protected $id;
    
    /* ... */
}
```
After this action every Item has new $id:
```php
class PostsController extends FOSRestController
{
    /**
     * [PUT] /posts/{slug} "put_post"
     *
     * @ParamConverter("post", converter="fos_rest.request_body")
     * @Rest\View()
     */
    public function putPostAction(Post $post) {
        
        $dm = $this->get('doctrine_mongodb')->getManager();
        $dm->merge($post);
        $dm->flush();

        return $post;
    }
}
```

**Comment by malarzm** [comment]
@websirnik could you please create a failing test case?

**Comment by malarzm** [comment]
ping @websirnik

**Comment by malarzm** [comment]
This should be fixed in `1.1.0` thanks to #1424
